HKMP/Game/Client/Entity/MantisLord.cs

```diff
@@ -0,0 +1,907 @@
+                }
+
+                if (state == ThroneState.Empty) {
+                    _throneFsm.ExecuteActions("Fighting", 1);
+                    _throneFsm.SetState("Fighting");
+                }
+
+                if (state == ThroneState.Defeated) {
+                    _throneFsm.ExecuteActions("Return", 2, 3);
+                    _throneFsm.SetState("Return");
+                }
+            }
+
+            // Now call the base class initializer, as we might have activated the mantis lord object
+            base.InternalInitializeAsSceneClient(baseStateIndex);
+        }
+
+        protected override void InternalSwitchToSceneHost() {
+            base.InternalSwitchToSceneHost();
+
+            RestoreAllTransitions(_throneFsm);
+
+            switch (_lastThroneAnimation) {
+                case ThroneAnimation.Stand:
+                    _throneFsm.SetState("Leave");
+                    break;
+                case ThroneAnimation.Leave:
+                    _throneFsm.SetState("Fighting");
+                    break;
+                case ThroneAnimation.Fighting:
+                    var healthManager = GameObject.GetComponent<HealthManager>();
+                    // We might have never received the defeated event, but the entity is dead already
+                    if (healthManager.GetIsDead()) {
+                        _throneFsm.SetState("Return Pause");
+                    }
+                    else {
+                        _throneFsm.SetState("Fighting");
+                    }
+                    break;
+            }
+        }
+
+        public override void UpdateAnimation(byte animationIndex, byte[] animationInfo) {
+            if (animationIndex <= (byte) Animation.Throw2 || animationIndex == 255) {
+                base.UpdateAnimation(animationIndex, animationInfo);
+                return;
+            }
+
+            var animation = (ThroneAnimation) animationIndex;
+
+            _lastThroneAnimation = animation;
+
+            Logger.Get().Info(this, $"Received animation: {animation}");
+
+            if (animation == ThroneAnimation.Stand) {
+                _throneFsm.ExecuteActions("Stand", 1);
+            }
+
+            if (animation == ThroneAnimation.Leave) {
+                _throneFsm.ExecuteActions("Leave", 1, 2);
+            }
+
+            if (animation == ThroneAnimation.Fighting) {
+                _throneFsm.ExecuteActions("Fighting", 1);
+            }
+
+            if (animation == ThroneAnimation.Return) {
+                _throneFsm.ExecuteActions("Return", 1, 2, 3);
+            }
+        }
+
+        protected enum ThroneAnimation {
+            Stand = Animation.Throw2 + 1,
+            Leave,
+            Fighting,
+            Return,
+        }
+
+        protected enum ThroneState {
+            Idle = 0,
+            Standing,
+            Empty,
+            Defeated,
+        }
+    }
+
+    public class MantisLord : MantisLordBase {
```

Comment:
Same as earlier comment about virtual method calls.
```suggestion
    public sealed class MantisLord : MantisLordBase {
```